### PR TITLE
select number of nodes to stake

### PR DESCRIPTION
**Pull request by sasurobert**
Improve selection of nodes to stake

### CODE REVIEW

**Review by raduchis — APPROVED** [review]

File: epochStart/metachain/systemSCs.go [hunk 1]
```diff
@@ -325,7 +328,12 @@ func (s *systemSCProcessor) unStakeNodesWithNotEnoughFunds(
 		return 0, err
 	}
 
-	return uint32(len(nodesToUnStake)), nil
+	nodesToStakeFromWaiting := uint32(len(nodesToUnStake))
```
Comment (AdoAdoAdo) [inline epochStart/metachain/systemSCs.go:331]: can we rename to activeNodesToUnstake ?

**Review by raduchis — DISMISSED** [review]
Maybe add 1 unit test

File: epochStart/metachain/systemSCs.go (hunk 1 above)
Comment (sasurobert) [reply epochStart/metachain/systemSCs.go:331]: these are actualy the number of nodes which will be staked from the waiting. nodes were unstaked previously on line 325

**Review by raduchis — DISMISSED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by SebastianMarian — DISMISSED** [review]

**Review by raduchis — DISMISSED** [review]

**Review by raduchis — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by raduchis — APPROVED** [review]
Tests passed, no new nodes were added to the waiting list when unStakeTokens and unStakedNodes from queue because of no enough funds

**Pull request merged**